foundations/foundation-metrics/src/main/java/io/servicecomb/foundation/metrics/output/servo/ServoMetricsFileOutput.java

```diff
@@ -0,0 +1,154 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicecomb.foundation.metrics.output.servo;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.log4j.Category;
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.Priority;
+import org.apache.log4j.spi.LoggingEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.servo.publish.BasicMetricFilter;
+import com.netflix.servo.publish.CounterToRateMetricTransform;
+import com.netflix.servo.publish.MetricObserver;
+import com.netflix.servo.publish.MonitorRegistryMetricPoller;
+import com.netflix.servo.publish.PollRunnable;
+import com.netflix.servo.publish.PollScheduler;
+
+import io.servicecomb.foundation.common.utils.RollingFileAppenderExt;
+import io.servicecomb.foundation.metrics.MetricsServoRegistry;
+import io.servicecomb.foundation.metrics.output.MetricsFileOutput;
+import io.servicecomb.serviceregistry.RegistryUtils;
+import io.servicecomb.serviceregistry.api.registry.Microservice;
+
+@Component
+public class ServoMetricsFileOutput extends MetricsFileOutput {
+
+  private static final Logger logger = LoggerFactory.getLogger(ServoMetricsFileOutput.class);
+
+  private final String applicationName;
+  private final ObjectMapper mapper = new ObjectMapper();
+  private final Map<String, RollingFileAppenderExt> metricsAppenders;
+  private final MetricsServoRegistry registry;
+  private String hostName;
+
+
+  @Autowired
+  public ServoMetricsFileOutput(MetricsServoRegistry registry) {
+    this.metricsAppenders = new HashMap<>();
+    this.registry = registry;
+    try {
+      InetAddress localHost = InetAddress.getLocalHost();
+      hostName = localHost.getHostName();
+    } catch (UnknownHostException e) {
+      e.printStackTrace();
+      hostName = "UnknownHost";
+    }
+
+    if (RegistryUtils.getServiceRegistry() != null) {
+      Microservice microservice = RegistryUtils.getMicroservice();
+      applicationName = String.join(".", microservice.getAppId(), microservice.getServiceName());
+    } else {
+      applicationName = String.join(".", hostName, "test");
```

Comment:
What's the mean of test,  if you just want to make the UT passed, it's not an acceptable reason.